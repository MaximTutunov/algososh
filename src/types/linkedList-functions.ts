export enum LinkedListFunctions {
    Waiting = '',
    AddToHead = 'AddToHead',
    AddToTail = 'AddToTail',
    DeleteFromHead = 'DeleteFromHead',
    DeleteFromTail = 'DeleteFromTail',
    AddByIndex = 'AddByIndex',
    DeleteByIndex = 'DeleteByIndex',
  }
  