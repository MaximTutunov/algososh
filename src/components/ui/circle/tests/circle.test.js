import renderer from "react-test-renderer";
import { Circle } from "../circle";
import {ElementStates} from "../../../../types";

describe("тестирование компонента Circle", () => {
  it("рендер компонента Circle без буквы", () => {
    const tree = renderer.create(<Circle />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("рендер компонента Circle с буквами", () => {
    const tree = renderer.create(<Circle letter="abc" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("рендер компонента Circle с head", () => {
    const tree = renderer.create(<Circle head={"Head"} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("рендер компонента Circle с React-component in head", () => {
    const tree = renderer
      .create(<Circle head={<div>ReactElement</div>} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("рендер компонента Circle с tail", () => {
    const tree = renderer.create(<Circle tail={"Tail"} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("рендер компонента Circle с React-component in tail", () => {
    const tree = renderer
      .create(<Circle tail={<div>ReactElement</div>} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("рендер компонента Circle с индексом", () => {
    const tree = renderer.create(<Circle index="1" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("рендер компонента Circle isSmall === true", () => {
    const tree = renderer.create(<Circle isSmall={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("рендер компонента Circle с ElementStates.Default", () => {
    const tree = renderer.create(<Circle state={ElementStates.Default}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("рендер компонента Circle с ElementStates.Changing", () => {
    const tree = renderer.create(<Circle state={ElementStates.Changing} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("рендер компонента Circle с ElementStates.Modified", () => {
    const tree = renderer.create(<Circle state={ElementStates.Modified} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
