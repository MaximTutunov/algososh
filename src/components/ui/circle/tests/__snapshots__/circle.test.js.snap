// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`тестирование компонента Circle рендер компонента Circle isSmall === true 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`тестирование компонента Circle рендер компонента Circle без буквы 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`тестирование компонента Circle рендер компонента Circle с ElementStates.Changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`тестирование компонента Circle рендер компонента Circle с ElementStates.Default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`тестирование компонента Circle рендер компонента Circle с ElementStates.Modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`тестирование компонента Circle рендер компонента Circle с React-component in head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div>
      ReactElement
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`тестирование компонента Circle рендер компонента Circle с React-component in tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div>
      ReactElement
    </div>
  </div>
</div>
`;

exports[`тестирование компонента Circle рендер компонента Circle с head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    Head
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`тестирование компонента Circle рендер компонента Circle с tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    Tail
  </div>
</div>
`;

exports[`тестирование компонента Circle рендер компонента Circle с буквами 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      abc
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`тестирование компонента Circle рендер компонента Circle с индексом 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;
